src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -204,36 +201,43 @@ private CompletableFuture<Void> statsFuture() {
 
   /**
    * @param isForce if true will flush regardless of other conditions
-   * @param timeDiffMillis Time in milliseconds since the last flush
+   * @param tablesToFlush list of tables to flush
+   * @param timeDiffMillis time difference in milliseconds
    * @return
    */
-  private CompletableFuture<Void> distributeFlush(boolean isForce, long timeDiffMillis) {
+  private CompletableFuture<Void> distributeFlush(
+      boolean isForce, Set<String> tablesToFlush, Long timeDiffMillis) {
     return CompletableFuture.runAsync(
         () -> {
-          logFlushTask(isForce, timeDiffMillis);
-          distributeFlushTasks();
-          this.isNeedFlush = false;
+          logFlushTask(isForce, tablesToFlush, timeDiffMillis);
+          distributeFlushTasks(tablesToFlush);
           this.lastFlushTime = System.currentTimeMillis();
```

Comment:
curious, why are we still maintaining this.lastFlushTime in this class? Can you update the comment where the member is declared and clarify how there's a per-table lastFlushTime in ChannelCache + how this is different from that?

Reply:
+1 This looks like it is only used for logging the time since the last flush but now the flush delta is per table

Reply:
Removed and logged `flushTimeDiffMill` per table